test/sh_do_test.sh

```diff
@@ -30,14 +30,36 @@ output_file=${RESULTS_DIR}/${script_name}.out
 echo " ... ${script_name}"
 rm -f ${output_file}
 
-./${script_name} > ${output_file}  2>&1
-if [ $? -ne 0 ]; then
-    # duplicate d_do_test output
-    echo >> ${output_file}
-    echo ============================== >> ${output_file}
-    echo Test ${script_name} failed >> ${output_file}
-
-    echo Test ${script_name} failed. The logged output:
-    cat ${output_file}
-    exit 1
-fi
+function finish {
+    # reset output stream
+    set +x
+    exec 1>&40
+    exec 2>&41
+
+    if [ $1 -ne 0 ]; then
+        echo ==============================
+        echo Test ${script_name} failed. The logged output:
+        cat ${output_file}
```

Comment:
General note, would be nice to see a lot more quoting. This line (and many many others) will fail if the variable contains spaces or some other special characters. ShellCheck helps greatly in identifying such moments.

Reply:
FYI, I think if tests contained spaces, there's alot of places that would fail.  But I suppose that's not a good reason to perpetuate bad practice.